src/AspNet.Security.OAuth.DingTalk/DingTalkAuthenticationHandler.cs

```diff
@@ -0,0 +1,153 @@
+﻿/*
+ * Licensed under the Apache License, Version 2.0 (http://www.apache.org/licenses/LICENSE-2.0)
+ * See https://github.com/aspnet-contrib/AspNet.Security.OAuth.Providers
+ * for more information concerning the license and the contributors participating to this project.
+ */
+
+using JetBrains.Annotations;
+using Microsoft.AspNetCore.Authentication;
+using Microsoft.AspNetCore.Authentication.OAuth;
+using Microsoft.AspNetCore.WebUtilities;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Security.Claims;
+using System.Text.Encodings.Web;
+using System.Text.Json;
+using System.Threading.Tasks;
+
+namespace AspNet.Security.OAuth.DingTalk
+{
+    public class DingTalkAuthenticationHandler : OAuthHandler<DingTalkAuthenticationOptions>
+    {
+        public DingTalkAuthenticationHandler(
+            [NotNull] IOptionsMonitor<DingTalkAuthenticationOptions> options,
+            [NotNull] ILoggerFactory logger,
+            [NotNull] UrlEncoder encoder,
+            [NotNull] ISystemClock clock)
+            : base(options, logger, encoder, clock)
+        {
+        }
+        protected override string BuildChallengeUrl(AuthenticationProperties properties, string redirectUri)
+        {
+            string stateValue = Options.StateDataFormat.Protect(properties);
+            var address = QueryHelpers.AddQueryString(Options.AuthorizationEndpoint, new Dictionary<string, string>
+            {
+                ["appid"] = Options.ClientId,
+                ["response_type"] = "code",
+                ["scope"] = FormatScope(),
+                ["redirect_uri"] = redirectUri,
+                ["state"] = stateValue
+            });
+            return address;
+        }
+        protected override async Task<OAuthTokenResponse> ExchangeCodeAsync([NotNull] OAuthCodeExchangeContext context)
+        {
+            using var payload = JsonDocument.Parse("{\"access_token\":\"" + context.Code + "\"}");
```

Comment:
Use a construct like this to create the new JSON document, rather than trying to manually create a correctly JSON encoded string using string formatting:

https://github.com/aspnet-contrib/AspNet.Security.OAuth.Providers/blob/a865eedad1ad0717fca85c42086a2cd3a34c7d8f/src/AspNet.Security.OAuth.StackExchange/StackExchangeAuthenticationHandler.cs#L122-L137

Reply:
Has been adjusted